#165416463 Swagger Api Documentation For Ireporter

nadralia opened the pull request:
**What does this PR do?**

Sets up Swagger Api Documentation

**Description of Issue to be completed?**

Added api documentation for the two endpoints (login and signup)


**How should this be manually tested?**

First thing you need to do is, re-install the requirements again:
` pip3 install -r requirements.txt`

Then Run the application 
` python3 manage.py runserver`

After running the app , to access the documentation go to:
 http://127.0.0.1:8000/docs/
None

**What are the relevant GitHub issues?**
#165416463

**Screenshots (if appropriate)**
None

**Questions:**

None

nadralia committed:
Ch(Setup Swagger): Api Documentation

Added drf_yasg dependency for swagger integration
changed the setting.py file
Added the urls for documentation called (docs)

[Finishes #165416463]

nadralia committed:
Ch(Setup Swagger): Api Documentation

Added drf_yasg dependency for swagger integration
changed the setting.py file
Added the urls for documentation called (docs)

[Finishes #165416463]

nadralia committed:
Ch(Setup Swagger): Api Documentation

Added drf_yasg dependency for swagger integration
changed the setting.py file
Added the urls for documentation called (docs)

[Finishes #165416463]

nadralia committed:
fix merge conflicts

JamesMudidi reviewed (approved):
Nice work with the documentation

kalsmic commented:
Well done @nadralia, however, I believe later on we might have to customize the request body for the login, at the moment it includes the email as well, and yet we are only using the username to log in.

ZackAtama committed:
ch(update readme): update readme to reflect app progress (#6)

* ft(user login): add the user login feature

the current application has no login feature
this commit helps registered users to login to the application

[Delivers #165344112]

* ch(update requirements): update requirements file

add gunicorn to the requirements file

* ch(update readme): update readme to reflect app progress

the current readme has no guidence to how the application can be run
update readme to show contributors how to run the application

[Finishes #165343938]

* mend

nadralia committed:
Merge branch 'ch-swagger-api-documentation-165416463' of https://github.com/knightedge-technologies/ireporter into ch-swagger-api-documentation-165416463

llwasampijja reviewed (approved):
Documentation is working fine!
